fixed places related to heap hard limit configs (#42869)
dotnet/docs: docs/core/runtime-config/garbage-collector.md

}
```

### Heap hard limit

- The heap hard limit is defined as the maximum commit size, in bytes, for the GC heap and GC bookkeeping.
- This setting only applies to 64-bit computers.

> - If this is not configured but the process is running in a memory constraint environment, ie, inside a container with a specified memory limit, a default value is set which is the greater of 20 MB or 75% of the memory limit on the container.

Reviewer: ```suggestion
- If this limit isn't configured but the process is running in a memory-constrained environment, that is, inside a container with a specified memory limit, a default value is set. That default is the greater of 20 MB or 75% of the memory limit on the container.
```

> [!TIP]
> If you're setting the option in *runtimeconfig.json*, specify a decimal value. If you're setting the option as an environment variable, specify a hexadecimal value. For example, to specify a heap hard limit of 200 mebibytes (MiB), the values would be 209715200 for the JSON file and 0xC800000 or C800000 for the environment variable.

### Heap hard limit percent

> - Specifies the heap hard limit as a percentage of the total physical memory. If the process is running in a memory constraint environment, ie, inside a container with a specified memory limit, the total physical memory is the memory limit; otherwise it's what's available on the machine.

Reviewer: ```suggestion
- Specifies the heap hard limit as a percentage of the total physical memory. If the process is running in a memory-constrained environment, that is, inside a container with a specified memory limit, the total physical memory is the memory limit; otherwise it's what's available on the machine.
```
